airbyte-server/src/main/java/io/airbyte/server/converters/ConfigConverter.java

```diff
@@ -0,0 +1,137 @@
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.server.converters;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.yaml.Yamls;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.validation.json.JsonSchemaValidator;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConfigConverter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigConverter.class);
+  private static final String CONFIG_FOLDER_NAME = "AirbyteConfig";
+
+  private final String version;
+  private final Path storageRoot;
+  private final JsonSchemaValidator jsonSchemaValidator;
+
+  public ConfigConverter(final String version, final Path storageRoot, final JsonSchemaValidator jsonSchemaValidator) {
+    this.version = version;
+    this.storageRoot = storageRoot;
+    this.jsonSchemaValidator = jsonSchemaValidator;
+  }
+
+  public ConfigConverter(final String version, final Path storageRoot) {
+    this(version, storageRoot, new JsonSchemaValidator());
+  }
+
+  private Path buildConfigPath(ConfigSchema schemaType) {
+    return storageRoot.resolve(CONFIG_FOLDER_NAME)
+        .resolve(String.format("%s.yaml", schemaType.toString()));
+  }
+
+  public <T> void writeConfigList(ConfigSchema schemaType, List<T> configList) throws IOException {
+    final Path configPath = buildConfigPath(schemaType);
+    Files.createDirectories(configPath.getParent());
+    Files.writeString(configPath, Yamls.serialize(new ConfigWrapper(version, configList)));
```

Comment:
i thought we said these files would just be a yaml list. i guess we can figure this out as we put stuff together.

Reply:
if we are going with the structure you're doing we need to find a way to read / write from these yamls files in a streaming fashion when the parent node is an object. i think i have figured this out for when the parent is a list but don't know how hard it will be to do if it's not.